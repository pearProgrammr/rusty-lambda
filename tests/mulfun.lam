a := (\ x . (\ y . x * y));
((a 1) 2);
b := (a 5);
b;
b 4;
