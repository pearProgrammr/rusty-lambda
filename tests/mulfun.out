Num(2)
Closure { env: {"x": Num(5), "a": Closure { env: {}, name: "x", func_term: Lambda { var_name: "y", expr: MathOp { opr: Multiply, t1: Var("x"), t2: Var("y") } } }}, name: "y", func_term: MathOp { opr: Multiply, t1: Var("x"), t2: Var("y") } }
Num(20)
